Kitodo/src/main/java/org/kitodo/production/services/data/ProcessService.java

```diff
@@ -1742,48 +1366,49 @@ public String getBaseType(int processId) throws DataException {
      *
      * @param processIds
      *            list of document IDs to retrieve and process.
-     * @return a map where the keys are document IDs and the values are their associated base types
-     */
-    public Map<Integer, String> getIdBaseTypeMap(List<Integer> processIds) throws DataException {
-        return fetchIdToBaseTypeMap(processIds);
-    }
-
-    /**
-     * Filter for correction / solution messages.
-     *
-     * @param lpe
-     *            List of process properties
-     * @return List of filtered correction / solution messages
+     * @return a map where the keys are document IDs and the values are their
+     *         associated base types
      */
-    private List<PropertyDTO> filterForCorrectionSolutionMessages(List<PropertyDTO> lpe) {
-        List<PropertyDTO> filteredList = new ArrayList<>();
-
-        if (lpe.isEmpty()) {
-            return filteredList;
-        }
-
-        List<String> translationList = Arrays.asList("Correction required", "Correction performed",
-            "Korrektur notwendig", "Korrektur durchgef\u00FChrt");
-
-        // filtering for correction and solution messages
-        for (PropertyDTO property : lpe) {
-            if (translationList.contains(property.getTitle())) {
-                filteredList.add(property);
+    public Map<Integer, String> getIdBaseTypeMap(List<Integer> processIds) {
+        File metadata = new File(ConfigCore.getKitodoDataDirectory());
+        return processIds.parallelStream().map(processId -> {
+            File metafile = new File(new File(metadata, processId.toString()), "meta.xml");
+            try {
+                String content = FileUtils.readFileToString(metafile, UTF_8);
+                Matcher type = TYPE_PATTERN.matcher(content);
+                return Pair.of(processId, type.find() ? type.group(1) : "");
+            } catch (IOException e) {
+                logger.catching(Level.ERROR, e);
+                return Pair.of(processId, e.getClass().getSimpleName());
             }
-        }
-        return filteredList;
+        }).collect(Collectors.toMap(Pair::getKey, Pair::getValue));
     }
```

Comment:
Have you tested that on slower storage (e.g. NFS) for more than 500 processes?

Reply:
Not only for 500 processes even for a few thousand processes as my collegues are creating excel lists with many entries. I think it is more useful to store the document type into the database instead of "grepping" it every time if the document type is needed.

Reply:
Option 3 would be to store it in the Seach Index again, as we do in 3.8. It should be possible to construct a projection query with Hibernate search querying for all the IDs in the structure tree at once and return the Type. And traverse the files as an emergency fallback.